packages/metrics/index.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type React from "react";
+import { getInjectionToken } from "@ogre-tools/injectable";
+
+export type ClusterOverviewUIBlock = {
+  id: string;
+  Component: React.ElementType;
+  orderNumber: number;
+};
+
+export const clusterOverviewUIBlockInjectionToken = getInjectionToken<ClusterOverviewUIBlock>({
+  id: "cluster-overview-ui-block-injection-token",
+});
+
+export type KubeObjectDetailMetricsComponentProps = {
+  object: any;
+};
+
+export type KubeObjectDetailMetrics = {
+  id: string;
+  Component: React.ComponentType<KubeObjectDetailMetricsComponentProps>;
+};
+
+export const podDetailsMetricsInjectionToken = getInjectionToken<KubeObjectDetailMetrics>({
+  id: "pod-details-metrics-injection-token",
+});
+
+export const deploymentDetailsMetricsInjectionToken = getInjectionToken<KubeObjectDetailMetrics>({
+  id: "deployment-details-metrics-injection-token",
+});
+
+export const nodeDetailsMetricsInjectionToken = getInjectionToken<KubeObjectDetailMetrics>({
+  id: "node-details-metrics-injection-token",
+});
+
+export const replicaSetDetailsMetricsInjectionToken = getInjectionToken<KubeObjectDetailMetrics>({
+  id: "replica-set-details-metrics-injection-token",
+});
+
+export const persistentVolumeClaimDetailsMetricsInjectionToken = getInjectionToken<KubeObjectDetailMetrics>({
+  id: "persistent-volume-claim-details-metrics-injection-token",
+});
+
+export const statefulSetDetailsMetricsInjectionToken = getInjectionToken<KubeObjectDetailMetrics>({
+  id: "stateful-set-details-metrics-injection-token",
+});
+
+export const namespaceDetailsMetricsInjectionToken = getInjectionToken<KubeObjectDetailMetrics>({
```

Comment:
Is it possible to make all of these bit more generic, so that we don't need multiple `injectionToken`s for same kind of stuff?

Reply:
I think it's useful to have separate tokens for each resource type. Maybe someone wants to implement metrics just for one resource but not all?

We don't implement metrics for all resource types ourselves.  The other thing is that `core` decides the placement of the metrics (at least for now), for example `Pod` details have both `Pod` metrics and `Container` metrics where Pod metrics are on top of the details and Container lower in the middle of the details component.